com.unity.render-pipelines.universal/Editor/AnimationClipUpgrader.cs

```diff
@@ -0,0 +1,645 @@
+        /// Without this mapping, the method makes inferences about how a material might have been upgraded.
+        /// Making these inferences is slower and possibly not sensitive to ambiguous upgrade paths.
+        /// </param>
+        /// <param name="filterFlags">
+        /// Optional flags to filter out clips that are used in ways that are not safe for upgrading.
+        /// </param>
+        public static void DoUpgradeAllClipsMenuItem(
+            IEnumerable<MaterialUpgrader> allUpgraders,
+            IReadOnlyDictionary<UID, MaterialUpgrader> knownUpgradePaths = default,
+            SerializedShaderPropertyUsage filterFlags = ~(SerializedShaderPropertyUsage.UsedByUpgraded | SerializedShaderPropertyUsage.UsedByNonUpgraded)
+        )
+        {
+            var clipPaths = AssetDatabase.FindAssets("t:AnimationClip")
+                .Select(p => (ClipPath)AssetDatabase.GUIDToAssetPath(p))
+                .ToArray();
+            DoUpgradeClipsMenuItem(clipPaths, allUpgraders, knownUpgradePaths, filterFlags);
+        }
+
+        static void DoUpgradeClipsMenuItem(
+            ClipPath[] clipPaths,
+            IEnumerable<MaterialUpgrader> allUpgraders,
+            IReadOnlyDictionary<UID, MaterialUpgrader> upgradePathsUsedByMaterials,
+            SerializedShaderPropertyUsage filterFlags
+        )
+        {
+            // exit early if no clips
+            if (clipPaths?.Length == 0)
+                return;
+
+            // display dialog box
+            var dialogMessage = L10n.Tr(
+                "Upgrading Material curves in AnimationClips assumes you have already upgraded Materials and shaders as needed. " +
+                "It also requires loading assets that use clips to inspect their usage, which can be a slow process. " +
+                "Which asset types do you want to check for usage?"
+            );
+            const int k_PrefabsOnly = 2;
+            const int k_Cancel = 1;
+            var selection = EditorUtility.DisplayDialogComplex(
+                L10n.Tr("Upgrade AnimationClips"),
+                dialogMessage,
+                L10n.Tr("Prefabs and Scenes"),
+                L10n.Tr("Do Not Upgrade AnimationClips"),
+                L10n.Tr("Prefabs Only")
+            );
+
+            // exit early if canceled
+            if (selection == k_Cancel)
+                return;
+
+            // only include scene paths if user requested it
+            var prefabPaths = AssetDatabase.FindAssets("t:Prefab")
+                .Select(p => (PrefabPath)AssetDatabase.GUIDToAssetPath(p))
+                .ToArray();
+            var scenePaths = selection == k_PrefabsOnly
+                ? Array.Empty<ScenePath>()
+                : AssetDatabase.FindAssets("t:Scene")
+                    .Select(p => (ScenePath)AssetDatabase.GUIDToAssetPath(p))
+                    .ToArray();
+
+            // TODO: Make following display progress bar?
+
```

Comment:
YEs I think we would need a progress bar here.

Reply:
done in fc2fded335cbb3628d6276c97752f231af3e6166.